### PR TITLE
Remove ray.tasks() from API.

### PR DESCRIPTION
**Pull request by robertnishihara**
This also removes some other unnecessary code.

### CODE REVIEW

**Comment by robertnishihara** [comment]
Requires @edoakes's approval before merging.

**Review by edoakes — COMMENTED** [review]
Beautiful! I believe we can also remove the `TaskSpec` and `Task` wrappers in `includes/task.pxi` and `includes/task.pxd` now. Could you try tearing those out too?

File: python/ray/includes/unique_ids.pxi
Comment (robertnishihara) [inline python/ray/includes/unique_ids.pxi]: @edoakes can you confirm that it's ok to remove this comment?

File: python/ray/tests/test_multi_node.py
Comment (robertnishihara) [inline python/ray/tests/test_multi_node.py]: Adding this test to replace `test_monitors.py` (which wasn't even being run anyway).

File: python/ray/includes/unique_ids.pxi
Comment (edoakes) [reply python/ray/includes/unique_ids.pxi]: This comments means we should be able to remove the `hasattr` check below and the related `in_core_worker` flag. Let's either remove those or leave the TODO modified to say those can be removed now.

**Review by edoakes — APPROVED** [review]
Looks great!

File: python/ray/includes/unique_ids.pxi
Comment (robertnishihara) [reply python/ray/includes/unique_ids.pxi:138]: Done. Initially tried removing the code, but ran into some errors, so decided to just update the comment.

Comment (robertnishihara) [reply python/ray/includes/unique_ids.pxi:138]: In particular, ran into the error
>AttributeError: 'Worker' object has no attribute 'core_worker'

here

**Pull request merged**
